pkg/module/sync/runner_impl.go

```diff
@@ -138,9 +138,11 @@ func (r *RunnerImpl) setupModule(module *common.Module, kyma *v1beta2.Kyma) erro
 	return nil
 }
 
-func (r *RunnerImpl) SyncModuleStatus(ctx context.Context, kyma *v1beta2.Kyma, modules common.Modules) {
+func (r *RunnerImpl) SyncModuleStatus(ctx context.Context, kyma *v1beta2.Kyma, modules common.Modules,
+	kymaMetrics *metrics.KymaMetrics,
```

Comment:
Can we create and pass an interface here to replace concrete KymaMetrics? So unit testing of `RunnerImpl` becomes a bit easier and also `SyncModuleStatus` does not need to know implementation details of kyma metrics.

Reply:
In unit tests we might want to assert that within this function and more likely within the `DeleteNoLongerExistingModuleStatus` function successively, a function of this metrics dependency  (in the current case `RemoveModuleStateMetrics`) is called or not called depending on the scenario. Testing this is easy when providing an interface, because it can be mocked and asserted/verified on, hence my recommendation.
Since it makes testing easier, decouples dependencies and it's not really a big change to swap this signature with an interface, I can see some need.